### PR TITLE
added example for views and forms

### PR DESCRIPTION
**Pull request by willoci**

### CODE REVIEW

File: views-and-forms-java/build.gradle [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    annotationProcessor "io.micronaut:micronaut-inject-java"
+    annotationProcessor "io.micronaut:micronaut-validation"
+    implementation platform("io.micronaut:micronaut-bom:$micronautVersion")
+    implementation "io.micronaut:micronaut-inject"
+    implementation "io.micronaut:micronaut-validation"
+    implementation "io.micronaut:micronaut-runtime"
+    implementation "io.micronaut:micronaut-http-server-netty"
+    implementation "io.micronaut:micronaut-http-client"
+    runtimeOnly "ch.qos.logback:logback-classic:1.2.3"
+    compile 'com.github.jknack:handlebars:4.1.0'
```
Comment (x80486) [inline views-and-forms-java/build.gradle]: `compileOnly` instead.

File: views-and-forms-java/build.gradle [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    compile 'io.micronaut:micronaut-views'
```
Comment (jameskleeh) [inline views-and-forms-java/build.gradle]: This should just be `micronaut-views-handlebars`. No need to add the specific dependency for `handlebars` unless you want to override the version for some reason

File: views-and-forms-java/src/main/java/views/and/forms/java/controllers/SurveyController.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+@Controller("/")
+public class SurveyController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SurveyController.class);
+
+    @Get
+    @View("home")
+    public FruitChoices home() {
+
+        LOG.info("Sending home page " );
```
Comment (jameskleeh) [inline views-and-forms-java/src/main/java/views/and/forms/java/controllers/SurveyController.java]: Should be wrapped with `if (LOG.isInfoEnabled())`

File: views-and-forms-java/src/test/java/views/and/forms/java/SurveyControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@MicronautTest
+public class SurveyControllerTest {
+
+    @Inject
+    SurveyController itemUnderTest;
```
Comment (jameskleeh) [inline views-and-forms-java/src/test/java/views/and/forms/java/SurveyControllerTest.java]: We shouldn't be injecting the controller here to test anything. Either invoke the endpoint with something like Geb to test the HTML or move the logic into a service layer and inject that

File: views-and-forms-java/src/main/java/views/and/forms/java/controllers/SurveyController.java (hunk 3 above)
Comment (x80486) [reply views-and-forms-java/src/main/java/views/and/forms/java/controllers/SurveyController.java]: Not needed these days, plus there is no string concatenation. An `if` clause here (and anywhere in reality these days, if parameterized messages are used) is not needed :wink:

**Comment by jameskleeh** [comment]
@x80486 I appreciate the feedback, however I disagree

File: views-and-forms-java/build.gradle [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+tasks.withType(JavaExec) {
+    classpath += configurations.developmentOnly
+    jvmArgs('-noverify', '-XX:TieredStopAtLevel=1', '-Dcom.sun.management.jmxremote')
+}
+
+webdriverBinaries {
+    chromedriver {
+        version = chromeDriverVersion
+        if (DefaultNativePlatform.currentOperatingSystem.isWindows()) {
```
Comment (jameskleeh) [inline views-and-forms-java/build.gradle:73]: This is an internal class. Use https://docs.gradle.org/current/javadoc/org/gradle/nativeplatform/platform/OperatingSystem.html instead

Comment (willoci) [reply views-and-forms-java/build.gradle]: @jameskleeh fixed in latest commit

**Pull request merged**